### PR TITLE
enable syntax highlighting on github

### PR DESCRIPTION
**Pull request by tudorbarascu**
Hi and thank you for your very nice work which I'm using.

As I noticed you wrote the **Readme** file with some markup I thought of helping so that it's better presented. You can take a look at the end result at https://github.com/tudorbarascu/oracle_fdw .

All the best,
Tudor

### CODE REVIEW

**Comment by laurenz** [comment]
Thanks for your effort.

I understand that this changes make the README optically more pleasing for Github users, but at the same time it might make it harder for plain text consumers.
Who knows if oracle_fdw will always remain on Github?

And certainly I don't want to drag along two versions of the README, one in plain text and one in markdown.

On the other hand, markdown is designed to be fine to read as plain text, so maybe I should accept such a change.

I don't like that you have marked code that is in its own line with `` ` `` like [here](https://github.com/laurenz/oracle_fdw/pull/128/commits/8ab644bdb298ba35baf6caa5444a229e96b39ad6#diff-afc074c8d9fe89dfbc9e007ea8c62ee3R34) or with ```` ``` ```` like [here](https://github.com/laurenz/oracle_fdw/pull/128/commits/8ab644bdb298ba35baf6caa5444a229e96b39ad6#diff-afc074c8d9fe89dfbc9e007ea8c62ee3R42). Could you indent such lines with four spaces instead?

I'll comment other quibbles that I have inline in your patches.

File: README.oracle_fdw [hunk 1]
```diff
@@ -141,47 +148,47 @@ Foreign server options
...
-- table (required)
+- **table** (required)
 
   The Oracle table name.  This name must be written exactly as it occurs in
   Oracle's system catalog, so normally consist of uppercase letters only.
 
   To define a foreign table based on an arbitrary Oracle query, set this
   option to the query enclosed in parentheses, e.g.
-     OPTIONS (table '(SELECT col FROM tab WHERE val = ''string'')')
+     `OPTIONS (table '(SELECT col FROM tab WHERE val = ''string'')')`
```
Comment (laurenz) [inline README.oracle_fdw]: This should be its own indented code block.

File: README.oracle_fdw [hunk 2]
```diff
@@ -141,47 +148,47 @@ Foreign server options
...
 
-- plan_costs (deprecated, optional, defaults to "off")
+- **plan_costs** (deprecated, optional, defaults to "off")
 
   This option will not work with Oracle 9i or older (see "Problems" below).
 
   If set to yes/on/true, Oracle's cost estimates will be used.  The problem is
   that Oracle gives good estimates for the result size, but not for the cost
-  of the execution (the only estimate you can get is "execution time" in second
+  of the execution (the only estimate you can get is **execution time** in second
```
Comment (laurenz) [inline README.oracle_fdw]: This should remain "execution time".

File: README.oracle_fdw [hunk 3]
```diff
@@ -141,47 +148,47 @@ Foreign server options
...
   granularity).
```
Comment (laurenz) [inline README.oracle_fdw]: I'd like to preserve this line break.

File: README.oracle_fdw [hunk 4]
```diff
@@ -194,39 +201,39 @@ Foreign table options
   influence on PostgreSQL query planning.
   This option is deprecated and will be removed in a future release.
 
-- max_long (optional, defaults to "32767")
+- **max_long** (optional, defaults to "32767")
 
-  The maximal length of any LONG or LONG RAW columns in the Oracle table.
+  The maximal length of any `LONG` or `LONG RAW` columns in the Oracle table.
```
Comment (laurenz) [inline README.oracle_fdw]: I'd like to preserve this line break.

File: README.oracle_fdw [hunk 5]
```diff
@@ -194,39 +201,39 @@ Foreign table options
...
   twice, so large values will consume a lot of memory.
-  If "max_long" is less than the length of the longest value retrieved,
-  you will receive the error message "ORA-01406: fetched column value was
-  truncated".
+  If `max_long` is less than the length of the longest value retrieved,
+  you will receive the error message `ORA-01406: fetched column value was
+  truncated`.
 
-- readonly (optional, defaults to "false")
+- **readonly** (optional, defaults to `false`)
```
Comment (laurenz) [inline README.oracle_fdw]: This should remain "false" like "off" or "32767" above.

File: README.oracle_fdw [hunk 6]
```diff
@@ -194,39 +201,39 @@ Foreign table options
...
   on tables that you do not wish to be changed, to be prepared for an upgrade
   to PostgreSQL 9.3 or later.
 
-- sample_percent (optional, defaults to 100)
+- **sample_percent** (optional, defaults to 100)
 
-  This option only influences ANALYZE processing and can be useful to ANALYZE
-  very large tables in a reasonable time.
+  This option only influences **ANALYZE** processing and can be useful to
+  **ANALYZE** very large tables in a reasonable time.
```
Comment (laurenz) [inline README.oracle_fdw]: Please don't make these bold &ndash; it is a good idea to have bold reserved for options in this chapter.

If you choose to put it in backticks, you should also do that in line 191 above.

File: README.oracle_fdw [hunk 7]
```diff
@@ -194,39 +201,39 @@ Foreign table options
...
 
   The value must be between 0.000001 and 100 and defines the percentage of
   Oracle table blocks that will be randomly selected to calculate PostgreSQL
-  table statistics.  This is accomplished using the "SAMPLE BLOCK (x)"
+  table statistics.  This is accomplished using the `SAMPLE BLOCK (x)`
   clause in Oracle.
 
-  ANALYZE will fail with ORA-00933 for tables defined with Oracle queries and
-  may fail with ORA-01446 for tables defined with complex Oracle views.
+  **ANALYZE** will fail with `ORA-00933` for tables defined with Oracle queries and
```
Comment (laurenz) [inline README.oracle_fdw]: Please also not bold here.

File: README.oracle_fdw [hunk 8]
```diff
@@ -239,7 +246,7 @@ Foreign table options
 Column options (from PostgreSQL 9.2 on)
 ---------------------------------------
 
-- key (optional, defaults to "false")
+- **key** (optional, defaults to "false")
 
   If set to yes/on/true, the corresponding column on the foreign Oracle table
   is considered a primary key column.
```
Comment (laurenz) [inline README.oracle_fdw]: I'd like to preserve this line break.

File: README.oracle_fdw [hunk 9]
```diff
@@ -255,8 +262,8 @@ Oracle permissions
 The Oracle user will obviously need CREATE SESSION privilege and the right
```
Comment (laurenz) [inline README.oracle_fdw]: Either put that in backticks or (better) remove the backticks from SQL commands and Oracle privileges below.

File: README.oracle_fdw [hunk 10]
```diff
@@ -265,7 +272,7 @@ oracle_fdw caches Oracle connections because it is expensive to create an
 Oracle session for each individual query.  All connections are automatically
 closed when the PostgreSQL session ends.
 
-The function oracle_close_connections() can be used to close all cached Oracle
+The function `oracle_close_connections()` can be used to close all cached Oracle
 connections.  This can be useful for long-running sessions that don't access
 foreign tables all the time and want to avoid blocking the resources needed
 by an open Oracle connection.
```
Comment (laurenz) [inline README.oracle_fdw]: I'd like to preserve this line break.

File: README.oracle_fdw [hunk 11]
```diff
@@ -331,10 +339,11 @@ TIMESTAMP WITH LOCAL TIME ZOME | date, timestamp, timestamptz, char,
 INTERVAL YEAR TO MONTH         | interval, char, varchar, text
 INTERVAL DAY TO SECOND         | interval, char, varchar, text
 MDSYS.SDO_GEOMETRY             | geometry (see "PostGIS support" below)
+```
 
-If a NUMBER is converted to a boolean, "0" means "false", everything else "true".
+If a NUMBER is converted to a boolean, `0` means `false`, everything else `true`.
```
Comment (laurenz) [inline README.oracle_fdw]: I believe double quotes are better here.

File: README.oracle_fdw [hunk 12]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
 WHERE conditions and ORDER BY clauses
 -------------------------------------
 
-PostgreSQL will use all applicaple parts of the WHERE clause as a filter
+PostgreSQL will use all applicaple parts of the `WHERE` clause as a filter
```
Comment (laurenz) [inline README.oracle_fdw]: It is not backticked in line 5. Maybe leave it that way.

File: README.oracle_fdw [hunk 13]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
-of WHERE clauses", can greatly reduce the number of rows retrieved from Oracle
+safely be translated to Oracle SQL.  This feature, also known as **push-down
+of WHERE clauses**, can greatly reduce the number of rows retrieved from Oracle
 and may enable Oracle's optimizer to choose a good plan for accessing the
 required tables.
 
-Similarly, ORDER BY clauses will be pushed down to Oracle from PostgreSQL 9.2
-on wherever possible.  Note that no ORDER BY conition that sorts by a character
+Similarly, `ORDER BY` clauses will be pushed down to Oracle from PostgreSQL 9.2
+on wherever possible.  Note that no `ORDER BY` condition that sorts by a character
```
Comment (laurenz) [inline README.oracle_fdw]: Maybe also better without backticks.

File: README.oracle_fdw [hunk 14]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
 
 Modifying foreign data
 ----------------------
 
-From PostgreSQL 9.3 on, oracle_fdw supports INSERT, UPDATE and DELETE on
+From PostgreSQL 9.3 on, oracle_fdw supports `INSERT`, `UPDATE` and `DELETE` on
 foreign tables.  This is allowed by default (also in databases upgraded
 from an earlier PostgreSQL release) and can be disabled by setting the
-"readonly" table option.
+`readonly` table option.
```
Comment (laurenz) [inline README.oracle_fdw]: Should be bold like other options.

File: README.oracle_fdw [hunk 15]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
+for columns with Oracle data types `LONG` and `LONG RAW` (Oracle doesn't support
+these data types in the `RETURNING` clause).
 
 Triggers on foreign tables are supported from PostgreSQL 9.4.
-Triggers defined with AFTER and FOR EACH ROW require that the foreign table
-has no columns with Oracle data type LONG or LONG RAW.  This is because
-such triggers make use of the RETURNING clause mentioned above.
+Triggers defined with `AFTER` and `FOR EACH ROW` require that the foreign table
+has no columns with Oracle data type `LONG` or `LONG RAW`.  This is because
+such triggers make use of the `RETURNING` clause mentioned above.
```
Comment (laurenz) [inline README.oracle_fdw]: I'd do away with the backticks here and in all the lines above.

File: README.oracle_fdw [hunk 16]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
-SAVEPOINT work as expected.  Prepared statements involving Oracle are
+Transactions are forwarded to Oracle, so `BEGIN`, `COMMIT`, `ROLLBACK` and
+`SAVEPOINT` work as expected.  Prepared statements involving Oracle are
 not supported.  See "Internals" for details.
 
 Since oracle_fdw uses serialized transactions, it is possible that
 data modifying statements lead to a serialization failure:
 
-  ORA-08177: can't serialize access for this transaction
+ `ORA-08177: can't serialize access for this transaction`
```
Comment (laurenz) [inline README.oracle_fdw]: Should be an indented code block.

File: README.oracle_fdw [hunk 17]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
 
 This can happen if concurrent transactions modify the table and gets more
 likely in long running transactions.  Such errors can be identified by their
-SQLSTATE (40001).  An application using oracle_fdw should retry transactions
+`SQLSTATE (40001)`. An application using oracle_fdw should retry transactions
```
Comment (laurenz) [inline README.oracle_fdw]: These backticks are fine!

File: README.oracle_fdw [hunk 18]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
 ANALYZE
 -------
 
-From PostgreSQL version 9.2 on, you can use ANALYZE to gather statistics
+From PostgreSQL version 9.2 on, you can use `ANALYZE` to gather statistics
 on a foreign table.  This is supported by oracle_fdw.
 Statistics will be used for query planning, and for many queries will
 result in good (and much faster) row count estimates even if the table option
-"plan_costs" is turned off as long as the statistics are accurate.
+`plan_costs` is turned off as long as the statistics are accurate.
```
Comment (laurenz) [inline README.oracle_fdw]: Bold like other options.

File: README.oracle_fdw [hunk 19]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
 
 Keep in mind that analyzing an Oracle foreign table will result in a full
-sequential table scan.  You can use the table option "sample_percent" to
+sequential table scan.  You can use the table option `sample_percent` to
```
Comment (laurenz) [inline README.oracle_fdw]: Bold like other options.

File: README.oracle_fdw [hunk 20]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
 ---------------
 
-The data type "geometry" is only available when PostGIS is installed.
+The data type `geometry` is only available when PostGIS is installed.
 
-The only supported geometry types are POINT, LINE, POLYGON, MULTIPOINT,
-MULTILINE and MULTIPOLYGON in two and three dimensions.
+The only supported geometry types are `POINT`, `LINE`, `POLYGON`,
+`MULTIPOINT`, `MULTILINE` and `MULTIPOLYGON` in two and three dimensions.
 Empty PostGIS geometries are not supported because they have no equivalent
```
Comment (laurenz) [inline README.oracle_fdw]: Those backticks might better be omitted as well.

File: README.oracle_fdw [hunk 21]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
-  ALL_TAB_COLUMNS.  That includes tables, views and materialized views,
+- `IMPORT FOREIGN SCHEMA` will create foreign tables for all objects found in
+  `ALL_TAB_COLUMNS`.  That includes tables, views and materialized views,
   but not synonyms.
 
-- There are two supported options for IMPORT FOREIGN SCHEMA:
-  - "case": controls case folding for table and column names during import.
+- There are two supported options for `IMPORT FOREIGN SCHEMA`:
+  - `case`: controls case folding for table and column names during import.
     The possible values are:
```
Comment (laurenz) [inline README.oracle_fdw]: Bold like other options?

File: README.oracle_fdw [hunk 22]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
-      "keep": leave the names as they are in Oracle, usually in upper case.
-      "lower": translate all table and column names to lower case.
-      "smart": only translate names that are all upper case in Oracle
+    - `keep`: leave the names as they are in Oracle, usually in upper case.
+    - `lower`: translate all table and column names to lower case.
+    - `smart`: only translate names that are all upper case in Oracle
                (this is the default).
```
Comment (laurenz) [inline README.oracle_fdw]: Bold like other options?

File: README.oracle_fdw [hunk 23]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
-  - "readonly" (boolean): controls if imported tables can be modified.
-    If set to "true", all imported tables are created with the foreign
-    table option "readonly" set to "true" (see "Options" above).
-    The default is "false".
+  - `readonly` (boolean): controls if imported tables can be modified.
+    If set to `true`, all imported tables are created with the foreign
+    table option `readonly` set to `true` (see "Options" above).
+    The default is `false`.
```
Comment (laurenz) [inline README.oracle_fdw]: Bold like other options (but "true" and "false" should remain in double quotes).

File: README.oracle_fdw [hunk 24]
```diff
@@ -625,39 +637,40 @@ The Oracle database must be prepared as follows:
...
-oracle_fdw sets the MODULE of the Oracle session to "postgres" and the
-ACTION to the backend process number.  This can help identifying the Oracle
-session and allows you to trace it with DBMS_MONITOR.SERV_MOD_ACT_TRACE_ENABLE.
+oracle_fdw sets the `MODULE` of the Oracle session to `postgres` and the
+`ACTION` to the backend process number.  This can help identifying the Oracle
+session and allows you to trace it with `DBMS_MONITOR.SERV_MOD_ACT_TRACE_ENABLE`.
 
 oracle_fdw uses Oracle's result prefetching to avoid unnecessary client-server
-round-trips.  The prefetch row count can be configured with the "prefetch"
+round-trips.  The prefetch row count can be configured with the `prefetch`
```
Comment (laurenz) [inline README.oracle_fdw]: Bold.

File: README.oracle_fdw [hunk 25]
```diff
@@ -343,143 +352,143 @@ Oracle or PostgreSQL.
...
-"readonly" table option.
+`readonly` table option.
 
-For UPDATE and DELETE to work, the columns corresponding to the primary
-key columns of the Oracle table must have the "key" column option set.
+For `UPDATE` and `DELETE` to work, the columns corresponding to the primary
+key columns of the Oracle table must have the `key` column option set.
 These columns are used to identify a foreign table row, so make sure that
-the option is set on *all* columns that belong to the primary key.
+the option is set on **all** columns that belong to the primary key.
```
Comment (laurenz) [inline README.oracle_fdw]: I'd prefer italics here.

File: README.oracle_fdw [hunk 26]
```diff
@@ -568,17 +579,17 @@ Instant Client, TNS_ADMIN if you have configuration files, etc.)
...
 To upgrade from an oracle_fdw version before 1.0.0, use
 
-  ALTER EXTENSION oracle_fdw UPDATE;
+ `ALTER EXTENSION oracle_fdw UPDATE;`
 
-Note that the extension version as shown by the psql command "\x" or the
-system catalog "pg_available_extensions" is *not* the installed version
-of oracle_fdw.  To get the oracle_fdw version, use the function "oracle_diag".
+Note that the extension version as shown by the psql command `\x` or the
+system catalog `pg_available_extensions` is **not** the installed version
```
Comment (laurenz) [inline README.oracle_fdw]: I'd prefer italics here.

File: README.oracle_fdw [hunk 27]
```diff
@@ -625,39 +637,40 @@ The Oracle database must be prepared as follows:
...
 table option and is set to 200 by default.
 
-Rather than using a PLAN_TABLE to explain an Oracle query (which would require
+Rather than using a `PLAN_TABLE` to explain an Oracle query (which would require
 such a table to be created in the Oracle database), oracle_fdw uses execution
-plans stored in the library cache.  For that, an Oracle query is "explicitly
-described", which forces Oracle to parse the query.  The hard part is to find
-the SQL_ID and CHILD_NUMBER of the statement in V$SQL because the SQL_TEXT
+plans stored in the library cache.  For that, an Oracle query is **explicitly
+described**, which forces Oracle to parse the query.  The hard part is to find
```
Comment (laurenz) [inline README.oracle_fdw]: I think italics would be better here.

File: README.oracle_fdw [hunk 28]
```diff
@@ -625,39 +637,40 @@ The Oracle database must be prepared as follows:
...
+the `SQL_ID` and `CHILD_NUMBER` of the statement in `V$SQL` because the `SQL_TEXT`
 column contains only the first 1000 bytes of the query.
 Therefore, oracle_fdw adds a comment to the query that contains an MD5 hash
-of the query text.  This is used to search in V$SQL.
-The actual execution plan or cost information is retrieved from V$SQL_PLAN.
+of the query text.  This is used to search in `V$SQL`.
+The actual execution plan or cost information is retrieved from `V$SQL_PLAN`.
 
-oracle_fdw uses transaction isolation level "serializable" on the Oracle side,
+oracle_fdw uses transaction isolation level **serializable** on the Oracle side,
```
Comment (laurenz) [inline README.oracle_fdw]: Italics or maybe better plain upper case.

File: README.oracle_fdw [hunk 29]
```diff
@@ -625,39 +637,40 @@ The Oracle database must be prepared as follows:
...
 which corresponds to PostgreSQL's repeatable read.  This is necessary because
```
Comment (laurenz) [inline README.oracle_fdw]: "repeatable read" should be like "serializable" above.

File: README.oracle_fdw [hunk 30]
```diff
@@ -683,15 +696,15 @@ characters", typically a normal or inverted question mark, by Oracle.
 You will get no warning or error messages.
 
 If you use a PostgreSQL database encoding that Oracle does not know
-(currently, these are EUC_CN, EUC_KR, LATIN10, MULE_INTERNAL, WIN874
-and SQL_ASCII), non-ASCII characters cannot be translated correctly.
-You will get a warning in this case, and the characters will be replaced
-by replacement characters as described above.
+(currently, these are `EUC_CN`, `EUC_KR`, `LATIN10`, `MULE_INTERNAL`,
+`WIN874` and `SQL_ASCII`), non-ASCII characters cannot be translated
```
Comment (laurenz) [inline README.oracle_fdw]: Maybe just leave these in upper case as well.

File: README.oracle_fdw [hunk 31]
```diff
@@ -683,15 +696,15 @@ characters", typically a normal or inverted question mark, by Oracle.
...
+correctly. You will get a warning in this case, and the characters
+will be replaced by replacement characters as described above.
 
-You can set the "nls_lang" option of the foreign data wrapper to force a
+You can set the `nls_lang` option of the foreign data wrapper to force a
```
Comment (laurenz) [inline README.oracle_fdw]: Bold.

File: README.oracle_fdw [hunk 32]
```diff
@@ -726,51 +741,53 @@ LDAP libraries
 --------------
 
 The Oracle client shared library comes with its own LDAP client
-implementation conforming to RFC 1823, so these functions have the same
+implementation conforming to **RFC 1823**, so these functions have the same
```
Comment (laurenz) [inline README.oracle_fdw]: Maybe a hyperlink rather than bold.

File: README.oracle_fdw [hunk 33]
```diff
@@ -726,51 +741,53 @@ LDAP libraries
...
 names as OpenLDAP's.  This will lead to a name collision when the PostgreSQL
-server was configured "--with-ldap".
+server was configured `--with-ldap`.
 
 The name collision will not be detected, because oracle_fdw is loaded at
-runtime, but trouble will happen if anybody calls an LDAP function.
-Typically, OpenLDAP is loaded first, so if Oracle calls an LDAP function
+runtime, but trouble will happen if anybody calls an **LDAP** function.
+Typically, OpenLDAP is loaded first, so if Oracle calls an **LDAP** function
 (for example if you use "directory naming" name resolution), the backend
```
Comment (laurenz) [inline README.oracle_fdw]: Not bold please.

File: README.oracle_fdw [hunk 34]
```diff
@@ -726,51 +741,53 @@ LDAP libraries
...
 On some platforms, you can force Oracle's client shared library to be loaded
-before the PostgreSQL server is started (LD_PRELOAD on Linux).  Then Oracle's
-LDAP functions should get used.  In that case, Oracle may be able to use
-LDAP functionality, but using LDAP from PostgreSQL will crash the backend.
+before the PostgreSQL server is started (`LD_PRELOAD` on Linux).  Then Oracle's
+**LDAP** functions should get used.  In that case, Oracle may be able to use
+**LDAP** functionality, but using **LDAP** from PostgreSQL will crash the backend.
 
-You cannot use LDAP functionality both in PostgreSQL and in Oracle, period.
+You cannot use **LDAP** functionality both in PostgreSQL and in Oracle, period.
```
Comment (laurenz) [inline README.oracle_fdw]: LDAP should not be bold here or above.

File: README.oracle_fdw [hunk 35]
```diff
@@ -726,51 +741,53 @@ LDAP libraries
...
 
 Serialization errors
 --------------------
 
 In Oracle 11.2 or above, inserting the first row into a newly created
 Oracle table with oracle_fdw will lead to a serialization error.
 
-This is because of an Oracle feature called "deferred segment creation" which
+This is because of an Oracle feature called **deferred segment creation** which
```
Comment (laurenz) [inline README.oracle_fdw]: Better leave double quotes.

File: README.oracle_fdw [hunk 36]
```diff
@@ -726,51 +741,53 @@ LDAP libraries
...
 defers allocation of storage space for a new table until the first row
 is inserted.  This causes a serialization failure with serializable
-transactions (see document 1285464.1 in Oracle's knowledge base).
+transactions (see document **1285464.1** in Oracle's knowledge base).
```
Comment (laurenz) [inline README.oracle_fdw]: Better not bold.

File: README.oracle_fdw [hunk 37]
```diff
@@ -726,51 +741,53 @@ LDAP libraries
...
-first error or creating the table with "SEGMENT CREATION IMMEDIATE".
+first error or creating the table with `SEGMENT CREATION IMMEDIATE`.
 
 Oracle bugs
 -----------
 
 This is a list of Oracle bugs that have affected oracle_fdw in the past.
 
-Bug 2728408 can cause "ORA-8177 cannot serialize access for this transaction"
+Bug **2728408** can cause `ORA-8177 cannot serialize access for this transaction`
```
Comment (laurenz) [inline README.oracle_fdw]: Better not bold.

File: README.oracle_fdw [hunk 38]
```diff
@@ -781,12 +798,12 @@ Oracle server 9.2 (install Patch Set 9.2.0.4 or better).
...
+ `SELECT oracle_diag('ora_serv');`
 
 in your problem report.
 If that causes an error, please also include the output of
 
-  SELECT oracle_diag();
+ `SELECT oracle_diag();`
 
 If you have a problem or question or any kind of feedback, the preferred
 option is to open an issue on GitHub:
```
Comment (laurenz) [inline README.oracle_fdw]: I'd like to preserve the line breaks before and after the URL.

**Comment by tudorbarascu** [comment]
I will make a rebase based on your comments, thank you for those.
Plain text users do suffer a little, indeed. My thinking is that those who use plain text can get around easily while newcomers are helped.

File: README.oracle_fdw [hunk 39]
```diff
@@ -131,7 +138,7 @@ software contains a CREATE FOREIGN DATA WRAPPER statement you can use.)
 Foreign server options
 ----------------------
 
-- dbserver (required)
+- **dbserver** (required)
 
   The Oracle database connection string for the remote database.
```
Comment (tudorbarascu) [inline README.oracle_fdw]: Not able to follow on which line break.

File: README.oracle_fdw [hunk 40]
```diff
@@ -141,47 +148,47 @@ Foreign server options
 User mapping options
 --------------------
 
-- user (required)
+- **user** (required)
 
   The Oracle user name for the session.
```
Comment (tudorbarascu) [inline README.oracle_fdw]: Not able to follow on which line break.

File: README.oracle_fdw [hunk 41]
```diff
@@ -9,33 +9,34 @@ oracle_fdw was written by Laurenz Albe <laurenz.albe@wien.gv.at>
...
-9. Support
+1. [Cookbook](#1-cookbook)
+2. [Objects created by the extension](#2-objects-created-by-the-extension)
+3. [Options](#3-options)
+4. [Usage](#4-usage)
+5. [Installation Requirements](#5-installation-requirements)
+6. [Installation](#6-installation)
+7. [Internals](#7-internals)
+8. [Problems](#8-problems)
+9. [Support](#9-support)
```
Comment (tudorbarascu) [inline README.oracle_fdw]: I tried various methods but I couldn't ditch the paragraph tag. To solve the consistency issue I added a line between each numbered row (not optimal but that's all I could do)

Comment (chrullrich) [inline README.oracle_fdw]: You can avoid the blank line by inserting a paragraph of text between the end of the list and the first heading below. Something like "For troubleshooting tips, see section 8."

**Comment by laurenz** [comment]
Thanks for the contribution!

**Pull request merged**